addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/data/SmokeDetector.java

```diff
@@ -76,48 +75,6 @@ public boolean isManualTestActive() {
     }
 
     @Override
-    public int hashCode() {
-        final int prime = 31;
-        int result = super.hashCode();
-        result = prime * result + ((batteryHealth == null) ? 0 : batteryHealth.hashCode());
-        result = prime * result + ((coAlarmState == null) ? 0 : coAlarmState.hashCode());
-        result = prime * result + (isManualTestActive ? 1231 : 1237);
-        result = prime * result + ((smokeAlarmState == null) ? 0 : smokeAlarmState.hashCode());
-        result = prime * result + ((uiColorState == null) ? 0 : uiColorState.hashCode());
-        return result;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (!super.equals(obj)) {
-            return false;
-        }
-        if (getClass() != obj.getClass()) {
-            return false;
-        }
-        SmokeDetector other = (SmokeDetector) obj;
-        if (batteryHealth != other.batteryHealth) {
-            return false;
-        }
-        if (coAlarmState != other.coAlarmState) {
-            return false;
-        }
-        if (isManualTestActive != other.isManualTestActive) {
-            return false;
-        }
-        if (smokeAlarmState != other.smokeAlarmState) {
-            return false;
-        }
-        if (uiColorState != other.uiColorState) {
-            return false;
-        }
-        return true;
-    }
-
-    @Override
     public String toString() {
         return "SmokeDetector [batteryHealth=" + batteryHealth + ", coAlarmState=" + coAlarmState + ", smokeAlarmState="
```

Comment:
Are you sure that no property of this class is `null`? You maybe want to use `StringBuilder` here.

Reply:
A commons-lang ReflectionToStringBuilder could also clean up all these generated toString() methods. But there is no other code in openHAB or ESH using it. Would there be a reason for this?

These toString() methods are mainly used for logging and generating a string representation of  objects while debugging. So the reflection overhead would generally not be an issue. It would solve potential bugs and the maintenance burden when additional properties are added.

Reply:
Maybe it does not play well with OSGi? According to the JavaDoc:
```
     * It uses <code>AccessibleObject.setAccessible</code> to gain access to private fields. This means that it will
     * throw a security exception if run under a security manager, if the permissions are not set up correctly. It is
     * also not as efficient as testing explicitly.
```

Reply:
I've gone for the StringBuilder approach anyhow for compatiblity reasons. But it would still be nice to know if a ReflectionToStringBuilder would cause issues. :-)

Reply:
I am not familiar with the ReflectionToStringBuilder. I try to avoid using to much external dependencies. But if you already use commons.lang feel free to test it.